pynbody/snapshot/snapshot_util.py

```diff
@@ -21,14 +21,42 @@ class ContainerWithPhysicalUnitsOption:
     that can be converted to physical units.
     """
     _autoconvert = None
+    _units_conversion_cache = {}
+
+    @classmethod
+    def _cached_unit_conversion(cls, from_unit, dims, ucut=3):
+        key = (
+            repr(from_unit._bases),
+            repr(from_unit._powers),
+            tuple(dims),
+            ucut,
+        )
```

Comment:
Would it be better to have units implement a hash, to prevent needing to know internal structure of the class at this point?

Reply:
I have implemented a `_hash_dimensions` method that returns a string unique to its _dimensions_, so that any two units having the same dimensions have the same hash.
```python
units.cm._hash_dimensions() == units.kpc._hash_dimensions() == units.Unit("3e8 m s**-1 yr")._hash_dimensions()
```